cudax/include/cuda/experimental/__detail/config.cuh

```diff
@@ -34,7 +34,16 @@ namespace cuda::experimental::__async
 #  define _CUDAX_ARTIFICIAL
 #endif
 
-#define _CUDAX_ALWAYS_INLINE _CCCL_FORCEINLINE _CUDAX_ARTIFICIAL _LIBCUDACXX_NODEBUG
+// _CUDAX_HIDDEN is just like _CCCL_HIDE_FROM_ABI but does not add the `inline` keyword.
+#if defined(_CCCL_COMPILER_ICC) // ICC has issues with visibility attributes on symbols with internal linkage
+#  define _CUDAX_HIDDEN
+#else // ^^^ _CCCL_COMPILER_ICC ^^^ / vvv !_CCCL_COMPILER_ICC vvv
+#  define _CUDAX_HIDDEN _CCCL_VISIBILITY_HIDDEN _CCCL_EXCLUDE_FROM_EXPLICIT_INSTANTIATION
+#endif // !_CCCL_COMPILER_ICC
+
+// _CUDAX_HIDDEN_INLINE force-inlines a function, marks its visibility as hidden, and causes debuggers to skip it.
+// This is useful for trivial internal functions that do dispatching or other plumbing work.
+#define _CUDAX_HIDDEN_INLINE _CUDAX_HIDDEN _CCCL_FORCEINLINE _CUDAX_ARTIFICIAL _LIBCUDACXX_NODEBUG
```

Comment:
Honestly, I would rather just drop that for ICC than making everything else worse. We do not care too much about that compiler and it is an optimization we can happily drop

Reply:
that's fine, but i'd rather drop ICC across all of CCCL in its own PR. i'd like to leave this for now.

Reply:
I did not mean to drop ICC, but to disable the hidden attribute for it

Reply:
i'm probably being dense, but the hidden attribute _is_ disabled for ICC. `_CUDAX_HIDDEN` expands to nothing there.

Reply:
We are going to get a deprecation warning for ICC with #2076. We can probably drop ICC at the next major release.